File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l,r=0,len(height)-1
        ma=0
        while(l<r):
            small=min(height[l],height[r])
            ma=max(ma,small*(r-l))
            if height[l]<height[r]:
                l+=1
            else:
                r-=1
        return ma
                
        
        

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<char,int> mp;
        int l=0;
        int ans=0;
        for(int i=0;i<s.size();i++){
            if(mp[s[i]]&&l<mp[s[i]]){
                l=mp[s[i]];                
            }
            mp[s[i]]=i+1;
            ans=max(ans,i-l+1);
        }    
        return ans;
    }
};
